hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/HoodieIncrSource.java

```diff
@@ -128,10 +130,16 @@ public static class Config {
         HoodieIncrSourceConfig.HOODIE_DROP_ALL_META_FIELDS_FROM_SOURCE.defaultValue();
   }
 
+  private final Map<String, String> readOpts = new HashMap<>();
+
   public HoodieIncrSource(TypedProperties props, JavaSparkContext sparkContext, SparkSession sparkSession,
                           SchemaProvider schemaProvider) {
     super(props, sparkContext, sparkSession, schemaProvider);
 
+    for (Object key : props.keySet()) {
+      readOpts.put(key.toString(), props.getString(key.toString()));
```

Comment:
Instead of putting every properties here, only extract "FILE_GROUP_READER_ENABLED" and "USE_NEW_HUDI_PARQUET_FILE_FORMAT" here?

Reply:
Fixed.